database-commons/src/main/java/io/cdap/plugin/db/sink/AbstractDBSink.java

```diff
@@ -296,8 +310,21 @@ private Schema inferSchema(Class<? extends Driver> driverClass) {
           inferredFields.addAll(getSchemaReader().getSchemaFields(rs));
         }
       } catch (SQLException e) {
-        throw new InvalidStageException("Error while reading table metadata", e);
-
+        // wrap exception to ensure SQLException-child instances not exposed to contexts w/o jdbc driver in classpath
+        String errorMessageWithDetails = String.format("Error while reading table metadata." +
+          "Error message: '%s'. Error code: '%s'. SQLState: '%s'", e.getMessage(), e.getErrorCode(), e.getSQLState());
+        String externalDocumentationLink = getExternalDocumentationLink();
+        if (!Strings.isNullOrEmpty(externalDocumentationLink)) {
+          if (!errorMessageWithDetails.endsWith(".")) {
+            errorMessageWithDetails = errorMessageWithDetails + ".";
+          }
+          errorMessageWithDetails = String.format("%s For more details, see %s", errorMessageWithDetails,
+            externalDocumentationLink);
+        }
+        throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+          e.getMessage(), errorMessageWithDetails, ErrorType.USER, false, ErrorCodeType.SQLSTATE,
+          e.getSQLState(), externalDocumentationLink, new SQLException(e.getMessage(),
+            e.getSQLState(), e.getErrorCode()));
```

Comment:
NIT : remove space

Reply:
This is auto formatted, with extra intendation since `e.getSQLState(), e.getErrorCode())` is part of nested function call to `new SQLException`